openpathsampling/shooting.py

```diff
@@ -34,9 +33,15 @@ def probability(self, snapshot, trajectory):
         else:
             return 0.0
 
-    def probability_ratio(self, snapshot, old_trajectory, new_trajectory):
+    def probability_ratio(self, snapshot, old_trajectory,
+                          new_trajectory, new_snapshot=None):
+        # TODO OPS 2.0: We should probabily alter the order and keyword names
+        # of this call
+        if new_snapshot is None:
+            NEW_SNAPSHOT_SELECTOR.warn(stacklevel=3)
```

Comment:
I notice you've been setting `stacklevel=3` in these -- should the default `stacklevel=2` be changed?

Reply:
Maybe. When I run py.test on these with `stacklevel=2` I get
```
openpathsampling/tests/test_external_engine.py::TestExternalEngine::test_in_shooting_move
  /home/sander/github_files/openpathsampling/openpathsampling/shooting.py:41: DeprecationWarning: new_snapshot=None; If snapshot has been copied or modified we can't find it in trial_trajectory. This call signature will update to (old_snapshot, old_trajectory, new_snapshot, new_trajectory) in OpenPathSampling 2.0.  Call with kwargs and use new_snapshot=old_snapshot if  old_snapshot is not copied or modified in new_traj
    NEW_SNAPSHOT_SELECTOR.warn(stacklevel=2)
```
which is a lot less useful (in my opinion) than with `stacklevel=3`:
```
openpathsampling/tests/test_external_engine.py::TestExternalEngine::test_in_shooting_move
  /home/sander/github_files/openpathsampling/openpathsampling/pathmover.py:823: DeprecationWarning: new_snapshot=None; If snapshot has been copied or modified we can't find it in trial_trajectory. This call signature will update to (old_snapshot, old_trajectory, new_snapshot, new_trajectory) in OpenPathSampling 2.0.  Call with kwargs and use new_snapshot=old_snapshot if  old_snapshot is not copied or modified in new_traj
    bias = self.selector.probability_ratio(
```

So we should definetly default to stacklevel=3 for deprecations were we want people to update their calls.

(As a side note, this is `stacklevel=1`, which is why the default might be slightly off:
```
  /home/sander/github_files/openpathsampling/openpathsampling/deprecations.py:62: DeprecationWarning: new_snapshot=None; If snapshot has been copied or modified we can't find it in trial_trajectory. This call signature will update to (old_snapshot, old_trajectory, new_snapshot, new_trajectory) in OpenPathSampling 2.0.  Call with kwargs and use new_snapshot=old_snapshot if  old_snapshot is not copied or modified in new_traj
    warnings.warn(self.message, category, stacklevel=stacklevel)
```